crates/re_query/src/util.rs

```diff
@@ -57,21 +57,35 @@ impl VisibleHistory {
         to: VisibleHistoryBoundary::Infinite,
     };
 
-    pub fn from(&self, cursor: TimeInt) -> TimeInt {
+    /// Do not use this to build a [`TimeRange`], use [`Self::time_range`].
+    fn from(&self, cursor: TimeInt) -> TimeInt {
         match self.from {
             VisibleHistoryBoundary::Absolute(value) => TimeInt::from(value),
             VisibleHistoryBoundary::RelativeToTimeCursor(value) => cursor + TimeInt::from(value),
             VisibleHistoryBoundary::Infinite => TimeInt::MIN,
         }
     }
 
-    pub fn to(&self, cursor: TimeInt) -> TimeInt {
+    /// Do not use this to build a [`TimeRange`], use [`Self::time_range`].
+    fn to(&self, cursor: TimeInt) -> TimeInt {
         match self.to {
             VisibleHistoryBoundary::Absolute(value) => TimeInt::from(value),
             VisibleHistoryBoundary::RelativeToTimeCursor(value) => cursor + TimeInt::from(value),
             VisibleHistoryBoundary::Infinite => TimeInt::MAX,
         }
     }
+
+    pub fn time_range(&self, cursor: TimeInt) -> TimeRange {
+        let mut from = self.from(cursor);
+        let mut to = self.to(cursor);
+
+        // TODO(#4993): visible time range UI can yield inverted ranges
```

Comment:
not much point in logging: this happens everytime you play around with the slider really